contrib/hashicorp/vault/vault_test.go

```diff
@@ -0,0 +1,281 @@
+package vault
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/mocktracer"
+
+	vault "github.com/hashicorp/vault/api"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestClient(t *testing.T) {
+	assert := assert.New(t)
+	mt := mocktracer.Start()
+	defer mt.Stop()
+
+	client, err := vault.NewClient(&vault.Config{HttpClient: NewHttpClient()})
+	assert.NoError(err)
+
+	assert.NotNil(client)
+	assert.Nil(err)
+}
+
+func TestClientError(t *testing.T) {
+	assert := assert.New(t)
+	mt := mocktracer.Start()
+	defer mt.Stop()
+
+	var config = &vault.Config{
+		HttpClient: NewHttpClient(),
+		Address:    "http://bad host.com",
+	}
+
+	client, err := vault.NewClient(config)
+	assert.Nil(client)
+	assert.Error(err)
+}
+
+const (
+	secretMountPath = "/ns1/ns2/secret"
+)
+
+func setupServer(t *testing.T) *httptest.Server {
+	storage := make(map[string]string)
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method == "PUT" {
+			fmt.Fprintln(w, "{}")
+			defer r.Body.Close()
+			bodyBytes, err := ioutil.ReadAll(r.Body)
+			body := string(bodyBytes)
+			if err != nil {
+				t.Error(err)
+				return
+			}
+			storage[r.URL.Path] = body
+		} else if r.Method == "GET" {
+			if value, ok := storage[r.URL.Path]; ok {
```

Comment:
I think it's nice to get rid of as much indentation as possible. I prefer the "[_Happy path to the left_](https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88)" approach and it seems to be the norm. Maybe it's nicer to do:
```go
value, ok := storage[r.URL.Path]
if !ok {
    http.Error(w, "No data for key.", http.StatusNotFound)
    return
}
// continue...
```